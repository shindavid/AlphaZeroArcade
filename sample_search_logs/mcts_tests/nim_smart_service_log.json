{
"graphs" : [
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 2,
    "Q" : [0.309601, 0.690399],
    "state" : "p1 @7"
    },
    {
    "index" : 1,
    "N" : 1,
    "Q" : [0.5, 0.5],
    "state" : "* @4"
    }
  ],
  "edges" : [
    {
    "index" : 2,
    "from" : 0,
    "to" : 1,
    "E" : 1,
    "action" : 2
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 3,
    "Q" : [0.373068, 0.626932],
    "state" : "p1 @7"
    },
    {
    "index" : 1,
    "N" : 2,
    "Q" : [0.5, 0.5],
    "state" : "* @4"
    },
    {
    "index" : 2,
    "N" : 1,
    "Q" : [0.880797, 0.119203],
    "state" : "p0 @2"
    }
  ],
  "edges" : [
    {
    "index" : 2,
    "from" : 0,
    "to" : 1,
    "E" : 2,
    "action" : 2
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 2,
    "E" : 1,
    "action" : 2
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 4,
    "Q" : [0.404801, 0.595199],
    "state" : "p1 @7"
    },
    {
    "index" : 1,
    "N" : 3,
    "Q" : [0.5, 0.5],
    "state" : "* @4"
    },
    {
    "index" : 2,
    "N" : 2,
    "Q" : [0.940399, 0.0596015],
    "state" : "p0 @2"
    },
    {
    "index" : 3,
    "N" : 1,
    "Q" : [1, 0],
    "state" : "* @0"
    }
  ],
  "edges" : [
    {
    "index" : 2,
    "from" : 0,
    "to" : 1,
    "E" : 3,
    "action" : 2
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 2,
    "E" : 2,
    "action" : 2
    },
    {
    "index" : 7,
    "from" : 2,
    "to" : 3,
    "E" : 1,
    "action" : 1
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 5,
    "Q" : [0.423841, 0.576159],
    "state" : "p1 @7"
    },
    {
    "index" : 1,
    "N" : 4,
    "Q" : [0.5, 0.5],
    "state" : "* @4"
    },
    {
    "index" : 2,
    "N" : 2,
    "Q" : [0.940399, 0.0596015],
    "state" : "p0 @2"
    },
    {
    "index" : 3,
    "N" : 1,
    "Q" : [1, 0],
    "state" : "* @0"
    },
    {
    "index" : 4,
    "N" : 1,
    "Q" : [0.880797, 0.119203],
    "state" : "p0 @3"
    }
  ],
  "edges" : [
    {
    "index" : 2,
    "from" : 0,
    "to" : 1,
    "E" : 4,
    "action" : 2
    },
    {
    "index" : 4,
    "from" : 1,
    "to" : 4,
    "E" : 1,
    "action" : 1
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 2,
    "E" : 2,
    "action" : 2
    },
    {
    "index" : 7,
    "from" : 2,
    "to" : 3,
    "E" : 1,
    "action" : 1
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 6,
    "Q" : [0.436534, 0.563466],
    "state" : "p1 @7"
    },
    {
    "index" : 1,
    "N" : 5,
    "Q" : [0.5, 0.5],
    "state" : "* @4"
    },
    {
    "index" : 2,
    "N" : 3,
    "Q" : [0.960266, 0.0397343],
    "state" : "p0 @2"
    },
    {
    "index" : 3,
    "N" : 2,
    "Q" : [1, 0],
    "state" : "* @0"
    },
    {
    "index" : 4,
    "N" : 1,
    "Q" : [0.880797, 0.119203],
    "state" : "p0 @3"
    }
  ],
  "edges" : [
    {
    "index" : 2,
    "from" : 0,
    "to" : 1,
    "E" : 5,
    "action" : 2
    },
    {
    "index" : 4,
    "from" : 1,
    "to" : 4,
    "E" : 1,
    "action" : 1
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 2,
    "E" : 3,
    "action" : 2
    },
    {
    "index" : 7,
    "from" : 2,
    "to" : 3,
    "E" : 2,
    "action" : 1
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 7,
    "Q" : [0.4456, 0.5544],
    "state" : "p1 @7"
    },
    {
    "index" : 1,
    "N" : 6,
    "Q" : [0.5, 0.5],
    "state" : "* @4"
    },
    {
    "index" : 2,
    "N" : 3,
    "Q" : [0.960266, 0.0397343],
    "state" : "p0 @2"
    },
    {
    "index" : 3,
    "N" : 2,
    "Q" : [1, 0],
    "state" : "* @0"
    },
    {
    "index" : 4,
    "N" : 2,
    "Q" : [0.940399, 0.0596015],
    "state" : "p0 @3"
    }
  ],
  "edges" : [
    {
    "index" : 2,
    "from" : 0,
    "to" : 1,
    "E" : 6,
    "action" : 2
    },
    {
    "index" : 4,
    "from" : 1,
    "to" : 4,
    "E" : 2,
    "action" : 1
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 2,
    "E" : 3,
    "action" : 2
    },
    {
    "index" : 7,
    "from" : 2,
    "to" : 3,
    "E" : 2,
    "action" : 1
    },
    {
    "index" : 10,
    "from" : 4,
    "to" : 3,
    "E" : 1,
    "action" : 2
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 8,
    "Q" : [0.4524, 0.5476],
    "state" : "p1 @7"
    },
    {
    "index" : 1,
    "N" : 7,
    "Q" : [0.5, 0.5],
    "state" : "* @4"
    },
    {
    "index" : 2,
    "N" : 4,
    "Q" : [0.970199, 0.0298007],
    "state" : "p0 @2"
    },
    {
    "index" : 3,
    "N" : 3,
    "Q" : [1, 0],
    "state" : "* @0"
    },
    {
    "index" : 4,
    "N" : 2,
    "Q" : [0.940399, 0.0596015],
    "state" : "p0 @3"
    }
  ],
  "edges" : [
    {
    "index" : 2,
    "from" : 0,
    "to" : 1,
    "E" : 7,
    "action" : 2
    },
    {
    "index" : 4,
    "from" : 1,
    "to" : 4,
    "E" : 2,
    "action" : 1
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 2,
    "E" : 4,
    "action" : 2
    },
    {
    "index" : 7,
    "from" : 2,
    "to" : 3,
    "E" : 3,
    "action" : 1
    },
    {
    "index" : 10,
    "from" : 4,
    "to" : 3,
    "E" : 1,
    "action" : 2
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 9,
    "Q" : [0.415378, 0.584622],
    "state" : "p1 @7"
    },
    {
    "index" : 1,
    "N" : 7,
    "Q" : [0.5, 0.5],
    "state" : "* @4"
    },
    {
    "index" : 2,
    "N" : 4,
    "Q" : [0.970199, 0.0298007],
    "state" : "p0 @2"
    },
    {
    "index" : 3,
    "N" : 3,
    "Q" : [1, 0],
    "state" : "* @0"
    },
    {
    "index" : 4,
    "N" : 2,
    "Q" : [0.940399, 0.0596015],
    "state" : "p0 @3"
    },
    {
    "index" : 5,
    "N" : 1,
    "Q" : [0.119203, 0.880797],
    "state" : "* @6"
    }
  ],
  "edges" : [
    {
    "index" : 0,
    "from" : 0,
    "to" : 5,
    "E" : 1,
    "action" : 0
    },
    {
    "index" : 2,
    "from" : 0,
    "to" : 1,
    "E" : 7,
    "action" : 2
    },
    {
    "index" : 4,
    "from" : 1,
    "to" : 4,
    "E" : 2,
    "action" : 1
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 2,
    "E" : 4,
    "action" : 2
    },
    {
    "index" : 7,
    "from" : 2,
    "to" : 3,
    "E" : 3,
    "action" : 1
    },
    {
    "index" : 10,
    "from" : 4,
    "to" : 3,
    "E" : 1,
    "action" : 2
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 10,
    "Q" : [0.385761, 0.614239],
    "state" : "p1 @7"
    },
    {
    "index" : 1,
    "N" : 7,
    "Q" : [0.5, 0.5],
    "state" : "* @4"
    },
    {
    "index" : 2,
    "N" : 4,
    "Q" : [0.970199, 0.0298007],
    "state" : "p0 @2"
    },
    {
    "index" : 3,
    "N" : 3,
    "Q" : [1, 0],
    "state" : "* @0"
    },
    {
    "index" : 4,
    "N" : 2,
    "Q" : [0.940399, 0.0596015],
    "state" : "p0 @3"
    },
    {
    "index" : 5,
    "N" : 2,
    "Q" : [0.119203, 0.880797],
    "state" : "* @6"
    },
    {
    "index" : 6,
    "N" : 1,
    "Q" : [0.5, 0.5],
    "state" : "p0 @4"
    }
  ],
  "edges" : [
    {
    "index" : 0,
    "from" : 0,
    "to" : 5,
    "E" : 2,
    "action" : 0
    },
    {
    "index" : 2,
    "from" : 0,
    "to" : 1,
    "E" : 7,
    "action" : 2
    },
    {
    "index" : 4,
    "from" : 1,
    "to" : 4,
    "E" : 2,
    "action" : 1
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 2,
    "E" : 4,
    "action" : 2
    },
    {
    "index" : 7,
    "from" : 2,
    "to" : 3,
    "E" : 3,
    "action" : 1
    },
    {
    "index" : 10,
    "from" : 4,
    "to" : 3,
    "E" : 1,
    "action" : 2
    },
    {
    "index" : 13,
    "from" : 5,
    "to" : 6,
    "E" : 1,
    "action" : 2
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 11,
    "Q" : [0.361528, 0.638472],
    "state" : "p1 @7"
    },
    {
    "index" : 1,
    "N" : 7,
    "Q" : [0.5, 0.5],
    "state" : "* @4"
    },
    {
    "index" : 2,
    "N" : 4,
    "Q" : [0.970199, 0.0298007],
    "state" : "p0 @2"
    },
    {
    "index" : 3,
    "N" : 3,
    "Q" : [1, 0],
    "state" : "* @0"
    },
    {
    "index" : 4,
    "N" : 2,
    "Q" : [0.940399, 0.0596015],
    "state" : "p0 @3"
    },
    {
    "index" : 5,
    "N" : 3,
    "Q" : [0.119203, 0.880797],
    "state" : "* @6"
    },
    {
    "index" : 6,
    "N" : 1,
    "Q" : [0.5, 0.5],
    "state" : "p0 @4"
    },
    {
    "index" : 7,
    "N" : 1,
    "Q" : [0.880797, 0.119203],
    "state" : "p0 @6"
    }
  ],
  "edges" : [
    {
    "index" : 0,
    "from" : 0,
    "to" : 5,
    "E" : 3,
    "action" : 0
    },
    {
    "index" : 2,
    "from" : 0,
    "to" : 1,
    "E" : 7,
    "action" : 2
    },
    {
    "index" : 4,
    "from" : 1,
    "to" : 4,
    "E" : 2,
    "action" : 1
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 2,
    "E" : 4,
    "action" : 2
    },
    {
    "index" : 7,
    "from" : 2,
    "to" : 3,
    "E" : 3,
    "action" : 1
    },
    {
    "index" : 10,
    "from" : 4,
    "to" : 3,
    "E" : 1,
    "action" : 2
    },
    {
    "index" : 11,
    "from" : 5,
    "to" : 7,
    "E" : 1,
    "action" : 0
    },
    {
    "index" : 13,
    "from" : 5,
    "to" : 6,
    "E" : 1,
    "action" : 2
    }
  ]
  }
]
}