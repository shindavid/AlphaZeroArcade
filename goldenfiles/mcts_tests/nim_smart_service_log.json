{
"graphs" : [
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 2,
    "Q" : [0.309601, 0.690399],
    "state" : "[4, 0]"
    },
    {
    "index" : 1,
    "N" : 1,
    "Q" : [0.119203, 0.880797],
    "state" : "[3, 1]"
    }
  ],
  "edges" : [
    {
    "index" : 0,
    "from" : 0,
    "to" : 1,
    "E" : 1,
    "action" : 0
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 3,
    "Q" : [0.246135, 0.753865],
    "state" : "[4, 0]"
    },
    {
    "index" : 1,
    "N" : 1,
    "Q" : [0.119203, 0.880797],
    "state" : "[3, 1]"
    },
    {
    "index" : 2,
    "N" : 1,
    "Q" : [0.119203, 0.880797],
    "state" : "[2, 1]"
    }
  ],
  "edges" : [
    {
    "index" : 0,
    "from" : 0,
    "to" : 1,
    "E" : 1,
    "action" : 0
    },
    {
    "index" : 1,
    "from" : 0,
    "to" : 2,
    "E" : 1,
    "action" : 1
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 4,
    "Q" : [0.214402, 0.785598],
    "state" : "[4, 0]"
    },
    {
    "index" : 1,
    "N" : 1,
    "Q" : [0.119203, 0.880797],
    "state" : "[3, 1]"
    },
    {
    "index" : 2,
    "N" : 1,
    "Q" : [0.119203, 0.880797],
    "state" : "[2, 1]"
    },
    {
    "index" : 3,
    "N" : 1,
    "Q" : [0.119203, 0.880797],
    "state" : "[1, 1]"
    }
  ],
  "edges" : [
    {
    "index" : 0,
    "from" : 0,
    "to" : 1,
    "E" : 1,
    "action" : 0
    },
    {
    "index" : 1,
    "from" : 0,
    "to" : 2,
    "E" : 1,
    "action" : 1
    },
    {
    "index" : 2,
    "from" : 0,
    "to" : 3,
    "E" : 1,
    "action" : 2
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 5,
    "Q" : [0.171522, 0.828478],
    "state" : "[4, 0]"
    },
    {
    "index" : 1,
    "N" : 2,
    "Q" : [0.0596015, 0.940399],
    "state" : "[3, 1]"
    },
    {
    "index" : 2,
    "N" : 1,
    "Q" : [0.119203, 0.880797],
    "state" : "[2, 1]"
    },
    {
    "index" : 3,
    "N" : 1,
    "Q" : [0.119203, 0.880797],
    "state" : "[1, 1]"
    },
    {
    "index" : 4,
    "N" : 1,
    "Q" : [0, 1],
    "state" : "[0, 0]"
    }
  ],
  "edges" : [
    {
    "index" : 0,
    "from" : 0,
    "to" : 1,
    "E" : 2,
    "action" : 0
    },
    {
    "index" : 1,
    "from" : 0,
    "to" : 2,
    "E" : 1,
    "action" : 1
    },
    {
    "index" : 2,
    "from" : 0,
    "to" : 3,
    "E" : 1,
    "action" : 2
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 4,
    "E" : 1,
    "action" : 2
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 6,
    "Q" : [0.142935, 0.857065],
    "state" : "[4, 0]"
    },
    {
    "index" : 1,
    "N" : 2,
    "Q" : [0.0596015, 0.940399],
    "state" : "[3, 1]"
    },
    {
    "index" : 2,
    "N" : 2,
    "Q" : [0.0596015, 0.940399],
    "state" : "[2, 1]"
    },
    {
    "index" : 3,
    "N" : 1,
    "Q" : [0.119203, 0.880797],
    "state" : "[1, 1]"
    },
    {
    "index" : 4,
    "N" : 1,
    "Q" : [0, 1],
    "state" : "[0, 0]"
    }
  ],
  "edges" : [
    {
    "index" : 0,
    "from" : 0,
    "to" : 1,
    "E" : 2,
    "action" : 0
    },
    {
    "index" : 1,
    "from" : 0,
    "to" : 2,
    "E" : 2,
    "action" : 1
    },
    {
    "index" : 2,
    "from" : 0,
    "to" : 3,
    "E" : 1,
    "action" : 2
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 4,
    "E" : 1,
    "action" : 2
    },
    {
    "index" : 7,
    "from" : 2,
    "to" : 4,
    "E" : 1,
    "action" : 1
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 7,
    "Q" : [0.122516, 0.877484],
    "state" : "[4, 0]"
    },
    {
    "index" : 1,
    "N" : 2,
    "Q" : [0.0596015, 0.940399],
    "state" : "[3, 1]"
    },
    {
    "index" : 2,
    "N" : 2,
    "Q" : [0.0596015, 0.940399],
    "state" : "[2, 1]"
    },
    {
    "index" : 3,
    "N" : 2,
    "Q" : [0.0596015, 0.940399],
    "state" : "[1, 1]"
    },
    {
    "index" : 4,
    "N" : 1,
    "Q" : [0, 1],
    "state" : "[0, 0]"
    }
  ],
  "edges" : [
    {
    "index" : 0,
    "from" : 0,
    "to" : 1,
    "E" : 2,
    "action" : 0
    },
    {
    "index" : 1,
    "from" : 0,
    "to" : 2,
    "E" : 2,
    "action" : 1
    },
    {
    "index" : 2,
    "from" : 0,
    "to" : 3,
    "E" : 2,
    "action" : 2
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 4,
    "E" : 1,
    "action" : 2
    },
    {
    "index" : 7,
    "from" : 2,
    "to" : 4,
    "E" : 1,
    "action" : 1
    },
    {
    "index" : 8,
    "from" : 3,
    "to" : 4,
    "E" : 1,
    "action" : 0
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 8,
    "Q" : [0.107201, 0.892799],
    "state" : "[4, 0]"
    },
    {
    "index" : 1,
    "N" : 3,
    "Q" : [0.0397343, 0.960266],
    "state" : "[3, 1]"
    },
    {
    "index" : 2,
    "N" : 2,
    "Q" : [0.0596015, 0.940399],
    "state" : "[2, 1]"
    },
    {
    "index" : 3,
    "N" : 2,
    "Q" : [0.0596015, 0.940399],
    "state" : "[1, 1]"
    },
    {
    "index" : 4,
    "N" : 2,
    "Q" : [0, 1],
    "state" : "[0, 0]"
    }
  ],
  "edges" : [
    {
    "index" : 0,
    "from" : 0,
    "to" : 1,
    "E" : 3,
    "action" : 0
    },
    {
    "index" : 1,
    "from" : 0,
    "to" : 2,
    "E" : 2,
    "action" : 1
    },
    {
    "index" : 2,
    "from" : 0,
    "to" : 3,
    "E" : 2,
    "action" : 2
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 4,
    "E" : 2,
    "action" : 2
    },
    {
    "index" : 7,
    "from" : 2,
    "to" : 4,
    "E" : 1,
    "action" : 1
    },
    {
    "index" : 8,
    "from" : 3,
    "to" : 4,
    "E" : 1,
    "action" : 0
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 9,
    "Q" : [0.0952899, 0.90471],
    "state" : "[4, 0]"
    },
    {
    "index" : 1,
    "N" : 3,
    "Q" : [0.0397343, 0.960266],
    "state" : "[3, 1]"
    },
    {
    "index" : 2,
    "N" : 3,
    "Q" : [0.0397343, 0.960266],
    "state" : "[2, 1]"
    },
    {
    "index" : 3,
    "N" : 2,
    "Q" : [0.0596015, 0.940399],
    "state" : "[1, 1]"
    },
    {
    "index" : 4,
    "N" : 2,
    "Q" : [0, 1],
    "state" : "[0, 0]"
    }
  ],
  "edges" : [
    {
    "index" : 0,
    "from" : 0,
    "to" : 1,
    "E" : 3,
    "action" : 0
    },
    {
    "index" : 1,
    "from" : 0,
    "to" : 2,
    "E" : 3,
    "action" : 1
    },
    {
    "index" : 2,
    "from" : 0,
    "to" : 3,
    "E" : 2,
    "action" : 2
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 4,
    "E" : 2,
    "action" : 2
    },
    {
    "index" : 7,
    "from" : 2,
    "to" : 4,
    "E" : 2,
    "action" : 1
    },
    {
    "index" : 8,
    "from" : 3,
    "to" : 4,
    "E" : 1,
    "action" : 0
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 10,
    "Q" : [0.0857609, 0.914239],
    "state" : "[4, 0]"
    },
    {
    "index" : 1,
    "N" : 3,
    "Q" : [0.0397343, 0.960266],
    "state" : "[3, 1]"
    },
    {
    "index" : 2,
    "N" : 3,
    "Q" : [0.0397343, 0.960266],
    "state" : "[2, 1]"
    },
    {
    "index" : 3,
    "N" : 3,
    "Q" : [0.0397343, 0.960266],
    "state" : "[1, 1]"
    },
    {
    "index" : 4,
    "N" : 2,
    "Q" : [0, 1],
    "state" : "[0, 0]"
    }
  ],
  "edges" : [
    {
    "index" : 0,
    "from" : 0,
    "to" : 1,
    "E" : 3,
    "action" : 0
    },
    {
    "index" : 1,
    "from" : 0,
    "to" : 2,
    "E" : 3,
    "action" : 1
    },
    {
    "index" : 2,
    "from" : 0,
    "to" : 3,
    "E" : 3,
    "action" : 2
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 4,
    "E" : 2,
    "action" : 2
    },
    {
    "index" : 7,
    "from" : 2,
    "to" : 4,
    "E" : 2,
    "action" : 1
    },
    {
    "index" : 8,
    "from" : 3,
    "to" : 4,
    "E" : 2,
    "action" : 0
    }
  ]
  },
  {
  "nodes" : [
    {
    "index" : 0,
    "N" : 11,
    "Q" : [0.0779644, 0.922036],
    "state" : "[4, 0]"
    },
    {
    "index" : 1,
    "N" : 4,
    "Q" : [0.0298007, 0.970199],
    "state" : "[3, 1]"
    },
    {
    "index" : 2,
    "N" : 3,
    "Q" : [0.0397343, 0.960266],
    "state" : "[2, 1]"
    },
    {
    "index" : 3,
    "N" : 3,
    "Q" : [0.0397343, 0.960266],
    "state" : "[1, 1]"
    },
    {
    "index" : 4,
    "N" : 3,
    "Q" : [0, 1],
    "state" : "[0, 0]"
    }
  ],
  "edges" : [
    {
    "index" : 0,
    "from" : 0,
    "to" : 1,
    "E" : 4,
    "action" : 0
    },
    {
    "index" : 1,
    "from" : 0,
    "to" : 2,
    "E" : 3,
    "action" : 1
    },
    {
    "index" : 2,
    "from" : 0,
    "to" : 3,
    "E" : 3,
    "action" : 2
    },
    {
    "index" : 5,
    "from" : 1,
    "to" : 4,
    "E" : 3,
    "action" : 2
    },
    {
    "index" : 7,
    "from" : 2,
    "to" : 4,
    "E" : 2,
    "action" : 1
    },
    {
    "index" : 8,
    "from" : 3,
    "to" : 4,
    "E" : 2,
    "action" : 0
    }
  ]
  }
]
}